xla/stream_executor/stream_executor_internal.h

```diff
@@ -335,6 +335,36 @@ class StreamExecutorInterface {
   // Performs linear search over alive GPU streams.
   virtual Stream* FindAllocatedStream(void* /*gpu_stream*/) { return nullptr; }
 
+  // The following methods access an internal log of some subset
+  // of arguments passed to other class methods.
+  // Used for testing/debugging purposes.
+
+  struct GemmCallRecord {
+    enum GemmType {
+      GC_Plain = 0,
+      GC_Strided_Batched = 1,
+      GC_Batched = 2,
+      GC_BlasLt = 3
+    };
+    GemmType op;
+    int flags;
+    uint64_t size1, size2;
+  };
+  // This may be expanded as necessary to trace other calls
+  using CallArgument = std::variant<GemmCallRecord>;
+
+  // Retrieves and clears internal argument logs.
+  virtual void GetArgumentLogs(std::vector<CallArgument>& logs) {}
+
+  // The default 0 means no logging.
+  enum SE_ArgumentLogging {
```

Comment:
There's no need to have the "SE_" prefix -- that's implied already from the namespace.  

Also prefer "ArgumentLoggingMode", since that's what this is.